internal/middleware/auth.go

```diff
@@ -57,3 +63,39 @@ func AuthenticateRebac(next http.Handler, jimm jujuapi.JIMM) http.Handler {
 		next.ServeHTTP(w, r.WithContext(ctx))
 	}), jimm)
 }
+
+// AuthenticateWithSessionTokenViaBasicAuth performs basic auth authentication and puts an identity in the request's context
```

Comment:
With the change in the name, the doc string should also reflect that we are (mis)using basic auth slightly and the session token is in the password with an empty user.